### DISCUSSION TITLE
lsp zero and coq?

### DISCUSSION THREAD

**Discussion by Stranget0** [VonHeikemen/lsp-zero.nvim#396] [Q&A]
So I am using lsp zero regularly with the cmp integration, but recently ive been looking more and more on using coq for autocompletion.
Is integrating it with lsp zero a good idea? Is it possible? How would you do it? Should I switch off from using lsp-zero?

**Comment by VonHeikemen** [comment]
You mean [coq_nvim](https://github.com/ms-jpq/coq_nvim)? Sure, you can use it. If I understand correctly you just have to add the capabilities settings to the server configuration. So, this should work.

```lua
local lsp_zero = require('lsp-zero')
local coq_settings = require('coq').lsp_ensure_capabilities()

lsp_zero.set_server_config({
  capabilities = coq_settings.capabilities,
})

-- then you call the language server you have installed
-- for example
require('lspconfig').rust_analyzer.setup({})
require('lspconfig').gleam.setup({})
```
